[CI] DefaultJdkTrustConfigTests testGetSystemTrustStoreWithNoSystemProperties failing

**Build Scans:**
- [elasticsearch-periodic #9142 / openjdk21_checkpart1_java-fips-matrix](https://gradle-enterprise.elastic.co/s/6hezt2g2auez4)
- [elasticsearch-periodic #9135 / openjdk21_checkpart1_java-fips-matrix](https://gradle-enterprise.elastic.co/s/vhy43tintnmhk)
- [elasticsearch-periodic #9128 / openjdk21_checkpart1_java-fips-matrix](https://gradle-enterprise.elastic.co/s/mvb7bs4cufmqo)
- [elasticsearch-periodic #9121 / openjdk21_checkpart1_java-fips-matrix](https://gradle-enterprise.elastic.co/s/yx4qu3kgiqzvq)
- [elasticsearch-periodic #9114 / openjdk21_checkpart1_java-fips-matrix](https://gradle-enterprise.elastic.co/s/vydgehimo22lw)

**Reproduction Line:**
```
./gradlew ":libs:ssl-config:test" --tests "org.elasticsearch.common.ssl.DefaultJdkTrustConfigTests.testGetNonPKCS11TrustStoreWithPasswordSet" -Dtests.seed=5D576828C42BEBB3 -Dtests.locale=sv-AX -Dtests.timezone=Pacific/Wake -Druntime.java=21 -Dtests.fips.enabled=true
```

**Applicable branches:**
9.1

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.common.ssl.DefaultJdkTrustConfigTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testGetSystemTrustStoreWithNoSystemProperties),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: not an empty array
     but: was []
```

**Issue Reasons:**
- [9.1] 5 consecutive failures in step openjdk21_checkpart1_java-fips-matrix
- [9.1] 5 failures in test testGetSystemTrustStoreWithNoSystemProperties (3.7% fail rate in 134 executions)
- [9.1] 5 failures in step openjdk21_checkpart1_java-fips-matrix (100.0% fail rate in 5 executions)
- [9.1] 5 failures in pipeline elasticsearch-periodic (100.0% fail rate in 5 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -572,12 +572,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.action.EsqlActionBreakerIT
   method: testTopNPushedToLucene
   issue: https://github.com/elastic/elasticsearch/issues/130505
-- class: org.elasticsearch.common.ssl.DefaultJdkTrustConfigTests
-  method: testGetSystemTrustStoreWithNoSystemProperties
-  issue: https://github.com/elastic/elasticsearch/issues/130517
-- class: org.elasticsearch.common.ssl.DefaultJdkTrustConfigTests
-  method: testGetNonPKCS11TrustStoreWithPasswordSet
-  issue: https://github.com/elastic/elasticsearch/issues/130519
 - class: org.elasticsearch.multiproject.test.CoreWithMultipleProjectsClientYamlTestSuiteIT
   method: test {yaml=indices.resolve_index/10_basic_resolve_index/Resolve index with hidden and closed indices}
   issue: https://github.com/elastic/elasticsearch/issues/130568

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -605,12 +605,6 @@ tests:
 - class: org.elasticsearch.reindex.ReindexFromRemoteBuildRestClientTests
   method: testBuildRestClient
   issue: https://github.com/elastic/elasticsearch/issues/130588
-- class: org.elasticsearch.http.netty4.Netty4HttpServerTransportTests
-  method: testRespondAfterServiceCloseWithClientCancel
-  issue: https://github.com/elastic/elasticsearch/issues/130590
-- class: org.elasticsearch.http.netty4.Netty4HttpServerTransportTests
-  method: testRespondAfterServiceCloseWithServerCancel
-  issue: https://github.com/elastic/elasticsearch/issues/130591
 - class: org.elasticsearch.xpack.esql.qa.multi_node.GenerativeIT
   method: test
   issue: https://github.com/elastic/elasticsearch/issues/130067

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -590,15 +590,6 @@ tests:
 - class: org.elasticsearch.multiproject.test.CoreWithMultipleProjectsClientYamlTestSuiteIT
   method: test {yaml=indices.resolve_index/10_basic_resolve_index/Resolve index with hidden and closed indices}
   issue: https://github.com/elastic/elasticsearch/issues/130568
-- class: org.elasticsearch.reindex.ReindexRestClientSslTests
-  method: testClientPassesClientCertificate
-  issue: https://github.com/elastic/elasticsearch/issues/130584
-- class: org.elasticsearch.reindex.ReindexRestClientSslTests
-  method: testClientFailsWithUntrustedCertificate
-  issue: https://github.com/elastic/elasticsearch/issues/130585
-- class: org.elasticsearch.reindex.ReindexRestClientSslTests
-  method: testClientSucceedsWithCertificateAuthorities
-  issue: https://github.com/elastic/elasticsearch/issues/130586
 - class: org.elasticsearch.xpack.esql.qa.multi_node.GenerativeIT
   method: test
   issue: https://github.com/elastic/elasticsearch/issues/130067

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -599,12 +599,6 @@ tests:
 - class: org.elasticsearch.reindex.ReindexRestClientSslTests
   method: testClientSucceedsWithCertificateAuthorities
   issue: https://github.com/elastic/elasticsearch/issues/130586
-- class: org.elasticsearch.reindex.ReindexFromRemoteBuildRestClientTests
-  method: testHeaders
-  issue: https://github.com/elastic/elasticsearch/issues/130587
-- class: org.elasticsearch.reindex.ReindexFromRemoteBuildRestClientTests
-  method: testBuildRestClient
-  issue: https://github.com/elastic/elasticsearch/issues/130588
 - class: org.elasticsearch.xpack.esql.qa.multi_node.GenerativeIT
   method: test
   issue: https://github.com/elastic/elasticsearch/issues/130067

--- a/test/framework/src/main/java/org/elasticsearch/entitlement/runtime/policy/TestPolicyManager.java
+++ b/test/framework/src/main/java/org/elasticsearch/entitlement/runtime/policy/TestPolicyManager.java
@@ -159,6 +159,8 @@ public class TestPolicyManager extends PolicyManager {
         "org.junit",
         "org.mockito",
         "net.bytebuddy", // Mockito uses this
+
+        "org.bouncycastle.jsse.provider" // Used in test code if FIPS is enabled, support more fine-grained config in ES-12128
     };
 
     @Override

